pkg/segment/writer/startree_test.go

```diff
@@ -769,6 +770,61 @@ func getTotalLevels(node *Node) int {
 	return 1 + max
 }
 
+func (builder *StarTreeBuilder) DeriveEnc(wip WipBlock, recGroupByValues [][]string) (map[string]map[string]uint32, error) {
+	numRecs := wip.blockSummary.RecCount
+	mp := make(map[string]map[string]uint32)
+	for colNum := range builder.groupByKeys {
+		mp[builder.groupByKeys[colNum]] = make(map[string]uint32)
+	}
+
+	for recNum := uint16(0); recNum < numRecs; recNum += 1 {
+		for i, grpValues := range recGroupByValues[recNum] {
+			mp[builder.groupByKeys[i]][grpValues] = builder.wipRecNumToColEnc[i][recNum]
+		}
+	}
```

Comment:
nit: this code is hard for me to understand, so it takes me a while to review. Here's the questions I had reading this:
1. What is `DeriveEnc` supposed to do? Perhaps a different name would make it more clear.
2. What does `mp` store? It looks like you wanted to name it `map` but that's a keyword so you went with `mp`. Just from `mp := make(map[string]map[string]uint32)`, I have very little idea what this is going to store

Reply:
Have renamed it and added a comment explaining what it does. TLDR: There is no specific way to get the reverse mapping of colValues to enc. So iterate over the records and manually create this reverse encoding map.